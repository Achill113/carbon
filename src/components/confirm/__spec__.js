import React from 'react';
import Portal from './../portal';
import Confirm from './confirm';
import { mount, shallow } from 'enzyme';

describe('Confirm', () => {
  let wrapper, onCancel, onConfirm;

  beforeEach(() => {
    onCancel = jasmine.createSpy('cancel');
    onConfirm = jasmine.createSpy('confirm');

    // TODO - 1007 - 'open' is main cause of failure here - why?
    wrapper = shallow(
      <Confirm
        onCancel={ onCancel }
        onConfirm={ onConfirm }
        title="Confirm title"
        subtitle='Confirm Subtitle'
        data-element='bar'
        data-role='baz'
      />
    );
  });

  describe('default snapshot', () => {
    it('renders as expected', () => {
      // TODO - 1007 - 'wrapper' undefined, scope issue?
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('confirmButtons', () => {
    describe('yes button', () => {
      it('triggers the onConfirm when the yes button is clicked', () => {
        // TODO - 1007 - 'wrapper' undefined, scope issue?
        wrapper.find('[data-element="confirm"]').simulate('click');
        expect(onConfirm).toHaveBeenCalled();
      });
    });

    describe('no button', () => {
      it('triggers the onCancel when the no button is clicked', () => {
        // TODO - 1007 - 'wrapper' undefined, scope issue?
        wrapper.find('[data-element="cancel"]').simulate('click');
        expect(onCancel).toHaveBeenCalled();
      });
    });

    describe('when custom labels are defined', () => {
      beforeEach(() => {
        wrapper = mount(
          <Confirm
            open
            onCancel={ onCancel }
            onConfirm={ onConfirm }
            confirmLabel='Delete'
            cancelLabel='Cancel'
          />
        );
      });

      it('returns a custom labels', () => {
        // TODO - 1007 - 'wrapper' undefined, scope issue?
        expect(wrapper.find('.carbon-button--primary').text()).toEqual('Delete');
        expect(wrapper.find('.carbon-button--secondary').text()).toEqual('Cancel');
      });
    });
  });
});
