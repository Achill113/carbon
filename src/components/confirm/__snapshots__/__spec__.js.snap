// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirm default snapshot renders as expected 1`] = `
<Confirm
  ariaRole="dialog"
  autoFocus={true}
  data-element="bar"
  data-role="baz"
  onCancel={[Function]}
  onConfirm={[Function]}
  open={true}
  showCloseIcon={false}
  size="extra-small"
  subtitle="Confirm Subtitle"
  title="Confirm title"
>
  <div
    className="carbon-dialog carbon-confirm"
    data-component="confirm"
    data-element="bar"
    data-role="baz"
    data-state="open"
  >
    <CSSTransitionGroup
      component="div"
      transitionAppear={false}
      transitionEnter={true}
      transitionEnterTimeout={500}
      transitionLeave={true}
      transitionLeaveTimeout={500}
      transitionName="modal"
    >
      <TransitionGroup
        childFactory={[Function]}
        component="div"
        transitionAppear={false}
        transitionEnter={true}
        transitionEnterTimeout={500}
        transitionLeave={true}
        transitionLeaveTimeout={500}
        transitionName="modal"
      >
        <div>
          <CSSTransitionGroupChild
            appear={false}
            enter={true}
            enterTimeout={500}
            key=".0"
            leave={true}
            leaveTimeout={500}
            name="modal"
          >
            <div
              aria-describedby="carbon-dialog-subtitle"
              aria-labelledby="carbon-dialog-title"
              className="carbon-dialog__dialog carbon-dialog__dialog--extra-small carbon-confirm__confirm"
              data-component="confirm"
              data-element="bar"
              data-role="baz"
              key=".0"
              onBlur={[Function]}
              role="dialog"
              style={
                Object {
                  "minHeight": undefined,
                }
              }
              tabIndex={0}
            >
              <div
                className="carbon-dialog__title carbon-dialog__title--has-subheader"
              >
                <Heading
                  divider={true}
                  separator={false}
                  subheader="Confirm Subtitle"
                  subtitleId="carbon-dialog-subtitle"
                  title="Confirm title"
                  titleId="carbon-dialog-title"
                >
                  <div
                    className="carbon-heading carbon-heading--has-subheader carbon-heading--has-divider"
                    data-component="heading"
                  >
                    <div
                      className="carbon-heading__header"
                    >
                      <div
                        className="carbon-heading__headers"
                      >
                        <div
                          className="carbon-heading__main-header"
                        >
                          <h1
                            className="carbon-heading__title"
                            data-element="title"
                            id="carbon-dialog-title"
                          >
                            Confirm title
                          </h1>
                        </div>
                        <div
                          className="carbon-heading__subheader"
                          data-element="subtitle"
                          id="carbon-dialog-subtitle"
                        >
                          Confirm Subtitle
                        </div>
                      </div>
                    </div>
                  </div>
                </Heading>
              </div>
              <div
                className="carbon-dialog__content"
              >
                <div
                  className="carbon-dialog__inner-content"
                  style={Object {}}
                >
                  <div
                    className="carbon-confirm__buttons clearfix"
                    key="confirm-buttons"
                  >
                    <div
                      className="carbon-confirm__button carbon-confirm__no"
                    >
                      <Button
                        as="secondary"
                        data-element="cancel"
                        disabled={false}
                        onClick={[Function]}
                        size="medium"
                        subtext=""
                        theme="blue"
                      >
                        <button
                          className="carbon-button carbon-button--secondary carbon-button--blue carbon-button--medium"
                          data-component="button"
                          data-element="cancel"
                          disabled={false}
                          onClick={[Function]}
                        >
                          No
                        </button>
                      </Button>
                    </div>
                    <div
                      className="carbon-confirm__button carbon-confirm__yes"
                    >
                      <Button
                        as="primary"
                        data-element="confirm"
                        disabled={false}
                        onClick={[Function]}
                        size="medium"
                        subtext=""
                        theme="blue"
                      >
                        <button
                          className="carbon-button carbon-button--primary carbon-button--blue carbon-button--medium"
                          data-component="button"
                          data-element="confirm"
                          disabled={false}
                          onClick={[Function]}
                        >
                          Yes
                        </button>
                      </Button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </CSSTransitionGroupChild>
        </div>
      </TransitionGroup>
    </CSSTransitionGroup>
    <CSSTransitionGroup
      component="div"
      transitionAppear={false}
      transitionEnter={true}
      transitionEnterTimeout={500}
      transitionLeave={true}
      transitionLeaveTimeout={500}
      transitionName="modal-background"
    >
      <TransitionGroup
        childFactory={[Function]}
        component="div"
        transitionAppear={false}
        transitionEnter={true}
        transitionEnterTimeout={500}
        transitionLeave={true}
        transitionLeaveTimeout={500}
        transitionName="modal-background"
      >
        <div>
          <CSSTransitionGroupChild
            appear={false}
            enter={true}
            enterTimeout={500}
            key=".0"
            leave={true}
            leaveTimeout={500}
            name="modal-background"
          >
            <div
              className="carbon-modal__background"
              key=".0"
            />
          </CSSTransitionGroupChild>
        </div>
      </TransitionGroup>
    </CSSTransitionGroup>
  </div>
</Confirm>
`;
